### ISSUE TITLE
Disable FXAA

### ISSUE THREAD

**Issue by BigheadSMZ** [HolographicWings/TOTK-Mods-collection#78]
There was an "exefs" version of disabling FXAA. I noticed it hasn't been updated since v1.1.1 anywhere, and now everywhere offers a "romfs" version that modifies "Bootup.Nin_NX_NVN.pack.zs". Is there a reason why this version is preferred? Does the "exefs" version really not do anything? I don't really prefer one version over the other as long as its working. It just seems to me that if the "exefs" version does actually work, wouldn't that be preferable since it wouldn't have the potential to conflict with other mods?

Below is the last version that updated, it was for TotK v1.1.1.
```
@nsobid-168DD518D925C7A327677286E72FEDA833314919

@flag print_values
@flag offset_shift 0x100

// FXAA (Probably does nothing, but makes sure FXAA is not enabled)
@enabled
027D149C 000080D2
0188CC50 000080D2
@stop
```

**Comment by HolographicWings** [comment]
We never had a solid proof of the Exefs to work
To be honest we are not even sure for the Romfs ones

**Comment by InfiniteParallel** [comment]
If the person who wrote the Exefs version could update it, despite it possibly not working, it would be appreciated. Loading romfs patches is problematic, to say the least.

**Comment by InfiniteParallel** [comment]
I might have been able to come up with something that will work for 1.1.2 using the exefs patch method. I can't really tell if it works or not, I guess the results are rather subtle.

[Disable FXAA 1.1.2.zip](https://github.com/HolographicWings/TOTK-Mods-collection/files/11594413/Disable.FXAA.1.1.2.zip)

**Comment by Dark-talon** [comment]
> We never had a solid proof of the Exefs to work To be honest we are not even sure for the Romfs ones

yes we are sure for the romfs ones, ive provided imgsli evidence for this already

**Comment by Dark-talon** [comment]
https://imgsli.com/MTgwNTA3

case closed

**Comment by Dark-talon** [comment]
> 

there is no downside to using romfs, because the only other files that conflict, are the resolution patches, that come in fxaa on vs off varities, and yuzu has already implemented fixes that dont have romfs load slow anymore

**Comment by BigheadSMZ** [comment]
> > 
> 
> there is no downside to using romfs, because the only other files that conflict, are the resolution patches, that come in fxaa on vs off varities, and yuzu has already implemented fixes that dont have romfs load slow anymore

Is that always going to be true though? I'm not sure what is all in that file since I haven't tried doing any actual modding myself. Is there any possibility future mods will conflict with it? (AKA, lots of interesting stuff to play with modders might enjoy modifying). I have no problem whatsoever with using romfs mods, I just know that over time the more they are used for graphics effect the higher the chance there will be conflicts until we have some kind of mod manager like BOTW has (BCML) that can automatically handle conflicts by setting priorities. 

> I might have been able to come up with something that will work for 1.1.2 using the exefs patch method. I can't really tell if it works or not, I guess the results are rather subtle.
> 
> [Disable FXAA 1.1.2.zip](https://github.com/HolographicWings/TOTK-Mods-collection/files/11594413/Disable.FXAA.1.1.2.zip)
> 
> I did test it with all my other graphical patches in place and nothing strange happened. No crashes, freezes, graphical corruption, etc... so, even if it doesn't work, it apparently, from my limited testing, shouldn't cause any trouble.

I'll test it and see if I can see any difference. I've been using the romfs version and even that has a very subtle difference.

**Comment by BigheadSMZ** [comment]
So after taking over a dozen screenshots and staring at blown up pixels, it appears that both the exefs and romfs versions work identically. It's honestly really hard to tell the difference between it being on and off, so I almost wonder why we're all going through the effort to figure it out haha. I guess scientific reasons are valid enough :). Ok, so here's some comparison shots. It's easiest to tell the difference between FXAA enabled and disabled on the rightmost edge of the stairwell when zoomed way in. 

https://imgsli.com/MTgzMTcy - 720p Mod - 1440p Resolution - 2x IR - Bilinear Filtering
https://imgsli.com/MTgzMTcz - 1008p Mod - 1440p Resolution - 2x IR - Bilinear Filtering
https://imgsli.com/MTgzMTc0 - 1008p Mod - 1440p Resolution - 2x IR - Nearest Filtering

I figured trying to get the game resolution as close to my monitors native resolution would make the best test case, so 720p mod with 2x internal resolution. But I also tried it with the 1008p mod at 2x resolution using both Bilinear and Nearest filters. I used Chuck's exefs version of the 1008p mod to keep it as basic as possible, but I also tested with the romfs version. I didn't post screenshots of this because the results were the same.

So the conclusion is that FXAA disabled via the exefs method does indeed work. This makes it worth keeping up-to-date IMO, because I'm already finding mods where the romfs version has conflicts (example: the "Better Fog" mod, link below).
https://www.nexusmods.com/legendofzeldatearsofthekingdom/mods/27?tab=description

Unfortunately, the exefs version of "Disable FSR Sharpening Shader" is in the same boat. While [InfiniteParallel](https://github.com/InfiniteParallel) managed to get an updated version of Disable FXAA for 1.1.2, Disable FSR Shader has not seen an update and currently remains exclusive to the romfs format. I'm not here to make demands, and I would keep these up-to-date myself if I had the required knowledge. I'm just going to put it out there that having multiple options are good. At least until someone inevitably takes on the task of better tooling that allows us to prioritize the contents of romfs mods.

**Comment by Dark-talon** [comment]
Great data! I think you may have labled the 2nd imgsli set backwards or something, as the fxaa off ones are blurrier and less jaggy, but the rest I think you demonstrated it very well.

And you are right, that the exefs code you have seems to be real, and it does override the romfs one. The romfs applies its setting at boot while the code applies durning the game so it will override it. Great demonstration.

Are all these just the 1.1.2 code? are you able to verify the codes work on 1.1.1 and 1.1.0 as well?

**Comment by InfiniteParallel** [comment]
I'm glad to see that the 1.1.2 update I made worked. It was just one location which is specific for just FXAA, so it's a simple mod that shouldn't conflict with anything else. I realized I made a small mistake on that 1.1.2 test version, I forgot to comment out the last line giving credit to the people who inspired the patch. I fixed that. Attached is the final version for 1.1.1 & 1.1.2.

[Disable FXAA 1.1.1.zip](https://github.com/HolographicWings/TOTK-Mods-collection/files/11627994/Disable.FXAA.1.1.1.zip)

[Disable FXAA 1.1.2.zip](https://github.com/HolographicWings/TOTK-Mods-collection/files/11627995/Disable.FXAA.1.1.2.zip)

I wasn't able to find the same value for 1.1.0, so I couldn't make a patch for that. If someone else can find it, please make it.

**Comment by BigheadSMZ** [comment]
Diving back in, I tested all game versions and all available disable FXAA mods. TL;DR, previously released FXAA disabled exefs mods do not work while the ones InfiniteParallel released above do work. This most likely explains the (probably does nothing) comments as they really don't seem to do anything. Also, the romfs version does not seem to work for v1.0.0, I even double backed and took two more sets of screenshots and I don't see any differences at all. I tried different configuration options, clearing my yuzu folder and starting over, to no avail. I could not get the romfs version working on 1.0.0, and I obviously used the version of the mod that was made for that version.

Test cases are same as before, 720p mod, 2x internal resolution, 1440p screen, bilinear filtering.

https://imgsli.com/MTgzNTg1 - v1.0.0
https://imgsli.com/MTgzNTg2 - v1.1.0
https://imgsli.com/MTgzNTg3 - v1.1.1

Both v1.0.0 and v1.1.0 are using the disable FXAA exefs mods that used to be included in the compilation on gbatemp by somerandompeople. For v1.1.1, I also included a comparison using the version by InfiniteParallel.

So.. conclusion time based on my observations:
v1.0.0 - neither romfs nor exefs versions work
v1.1.0 - romfs version works, no working exefs version
v1.1.1 - romfs version works, InfiniteParallel's exefs version works
v1.1.2 - romfs version works, InfiniteParallel's exefs version works

**Comment by igoticecream** [comment]
> 

1.1.2 patch uses the build id of 1.1.1

**Comment by BigheadSMZ** [comment]
Thanks to @Father_Of_Egg on reddit, there is now versions of the code for all four versions of TotK. I personally tested each of them and all of them work identically to each other and the romfs versions. I think it would be a good idea at this point to add it to the modpack as an alternative to the romfs versions as there is now several mods that the romfs versions conflict with. Eventually this won't be a problem when someone releases a mod manager that can merge files, but for now the problem exists. There at least seems to be some demand for it, in 18 hours after sharing it accumulated over 600 downloads. Anyway, here is some links to all four exefs versions.

[Disable FXAA (exefs) 1.1.0-1.1.2](https://www.mediafire.com/file/fknpj20ab8gsjzr/Disable_FXAA.zip/file) - Download Link
[Disable FXAA (exefs) 1.1.0-1.1.2](https://www.pixeldrain.com/u/MSNceGQG) - Mirror Link

So I can safely say this "issue" can be concluded. I originally set out to see if anyone could update the previous FXAA codes for 1.1.2 and it turns out they didn't even work to begin with. Then some working codes appeared by InfiniteParallel for 1.1.1 and 1.1.2, and now some working codes for 1.0.0 and 1.1.0 appeared by Father_of_Egg. Now it is up to everyone else if they want to use/distribute them. Thanks to everyone involved.

If this modpack does end up including them, feel free to make any necessary edits to the pchtxt files. The credits section for example, I have no idea who is responsible for what at this point so I just included everyone involved. When I first uploaded this, I accidentally included the 1.1.2 version with the wrong ID. When I was testing it using the first uploaded version for 1.1.2, the ID was correct so I didn't notice it wasn't correct in the second upload. Anyway, everything should be good to go in the files above.
